pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -309,18 +377,18 @@ func (r *ReconcileOneAgent) updateCR(instance *dynatracev1alpha1.OneAgent) error
 
 	newStatus := instance.Status
 
-	// Rather than sending the existing value, the .status.items map also gets replaced in-place, so we send a
-	// dummy object to avoid modifying it.
-	instance.Status = dynatracev1alpha1.OneAgentStatus{}
-
-	if err := r.client.Update(context.TODO(), instance); err != nil {
+	instance.Status = newStatus
+	r.logger.Info("Updating status")
+	// Now, with this call we do update the Status section to the new value.
+	if err := r.client.Status().Update(context.TODO(), instance); err != nil {
 		return err
 	}
 
-	instance.Status = newStatus
-
-	// Now, with this call we do update the Status section to the new value.
-	return r.client.Status().Update(context.TODO(), instance)
+	// Rather than sending the existing value, the .status.items map also gets replaced in-place, so we send a
+	// dummy object to avoid modifying it.
+	instance.Status = dynatracev1alpha1.OneAgentStatus{}
+	r.logger.Info("Updating spec")
+	return r.client.Update(context.TODO(), instance)
```

Comment:
Just for confirmation,

`.Status().Update()` doesn't replace `.spec` to the old values, right? - `.Update()` does modify `.status`, which is why I ask.